src/apis/auth.ts

```diff
@@ -93,111 +102,46 @@ export class Auth extends HttpClient {
 		super(env.PUBLIC_OPEN_REGISTRY_BACKEND_URL + '/auth');
 	}
 
-	public BrowserSupportsWebAuthN = (): boolean => {
-		if (window.PublicKeyCredential) {
-			return true;
-		}
-
-		return false;
-	};
-
 	public WebAuthNBeginRegister = async (username: string, email: string) => {
-		if (!this.BrowserSupportsWebAuthN()) {
+		if (!supported()) {
 			return {
 				error: 'Browser does not support WebAuthN'
 			};
 		}
 
 		const body = { username, email };
-
 		const path = 'webauthn/registration/begin';
-		const resp = await this.http.post(path, body);
+		const { error, data, status } = await this.http.post(path, body);
+		if (status !== 200) {
+			return { error, data, status };
+		}
 
-		const finishResponse = await this.WebAuthNFinishRegister(username, resp.data.options);
+		const options = (data as { options: CredentialCreationOptionsJSON }).options;
+		const finishResponse = await this.WebAuthNFinishRegister(username, options);
 		return finishResponse;
 	};
 
-	private bufferDecode = (value) => {
-		return Uint8Array.from(atob(value), (c) => c.charCodeAt(0));
-	};
-
-	private bufferEncode = (value: any) => {
-		return btoa(String.fromCharCode.apply(null, new Uint8Array(value)))
-			.replace(/\+/g, '-')
-			.replace(/\//g, '_')
-			.replace(/=/g, '');
-	};
-
-	public WebAuthNFinishRegister = async (username: string, credentialCreationOptions: any) => {
-		credentialCreationOptions.publicKey.challenge = this.bufferDecode(
-			credentialCreationOptions.publicKey.challenge
-		);
-		credentialCreationOptions.publicKey.user.id = this.bufferDecode(
-			credentialCreationOptions.publicKey.user.id
-		);
-
-		const credential = (await window.navigator.credentials.create({
-			publicKey: credentialCreationOptions.publicKey
-		})) as PublicKeyCredential;
-
-		const response = <AuthenticatorAttestationResponse>credential.response;
-		const attestationObject = response.attestationObject;
-		const clientDataJSON = response.clientDataJSON;
-		const rawId = credential.rawId;
-
-		const body = {
-			id: credential.id,
-			rawId: this.bufferEncode(rawId),
-			type: credential.type,
-			response: {
-				attestationObject: this.bufferEncode(attestationObject),
-				clientDataJSON: this.bufferEncode(clientDataJSON)
-			}
-		};
-
+	public WebAuthNFinishRegister = async (
+		username: string,
+		credentialCreationOptions: CredentialCreationOptionsJSON
+	) => {
+		const options = parseCreationOptionsFromJSON(credentialCreationOptions);
 		const path = `/webauthn/registration/finish?username=${username}`;
-		const resp = await this.http.post(path, body);
-		return resp;
+		const response = await create(options);
+		return await this.http.post(path, response);
 	};
 
 	public WebAuthNBeginLogin = async (username: string) => {
```

Comment:
@guacamole I think we should also do a check here if webauthn is supported? The same way we do it in `BeginRegister`